llama_stack/providers/adapters/safety/together/together.py

```diff
@@ -21,48 +50,42 @@ async def initialize(self) -> None:
     async def run_shield(
         self, shield_type: str, messages: List[Message], params: Dict[str, Any] = None
     ) -> RunShieldResponse:
-        if shield_type != "llama_guard":
-            raise ValueError(f"shield type {shield_type} is not supported")
-
-        provider_data = get_request_provider_data()
 
         together_api_key = None
-        if provider_data is not None:
-            if not isinstance(provider_data, TogetherProviderDataValidator):
-                raise ValueError(
-                    'Pass Together API Key in the header X-LlamaStack-ProviderData as { "together_api_key": <your api key>}'
-                )
-
-            together_api_key = provider_data.together_api_key
-        if not together_api_key:
-            together_api_key = self.config.api_key
+        provider_data = get_request_provider_data()
+        if provider_data is None or not provider_data.together_api_key:
+            raise ValueError(
+                'Pass Together API Key in the header X-LlamaStack-ProviderData as { "together_api_key": <your api key>}'
+            )
+        together_api_key = provider_data.together_api_key
 
-        if not together_api_key:
-            raise ValueError("The API key must be provider in the header or config")
+        model_name = shield_type_to_model_name(shield_type)
 
         # messages can have role assistant or user
         api_messages = []
         for message in messages:
             if message.role in (Role.user.value, Role.assistant.value):
                 api_messages.append({"role": message.role, "content": message.content})
 
-        violation = await get_safety_response(together_api_key, api_messages)
+        violation = await get_safety_response(
+            together_api_key, model_name, api_messages
+        )
         return RunShieldResponse(violation=violation)
 
 
 async def get_safety_response(
-    api_key: str, messages: List[Dict[str, str]]
+    api_key: str, model_name: str, messages: List[Dict[str, str]]
 ) -> Optional[SafetyViolation]:
     client = Together(api_key=api_key)
-    response = client.chat.completions.create(
-        messages=messages, model="meta-llama/Meta-Llama-Guard-3-8B"
-    )
+    response = client.chat.completions.create(messages=messages, model=model_name)
     if len(response.choices) == 0:
         return None
 
     response_text = response.choices[0].message.content
     if response_text == "safe":
-        return None
+        return SafetyViolation(
```

Comment:
this is not right! if there is no violation you should not return a Violation object?